packages/components/src/modifiers/hds-clipboard.ts

```diff
@@ -129,15 +136,35 @@ export const copyToClipboard = async (text, target) => {
 // because it's quite simple in its logic, and doesn't require injecting services
 // see: https://github.com/ember-modifier/ember-modifier#function-based-modifiers
 
-export default modifier((element, positional, named) => {
+export default modifier((element, _positional, named) => {
   assert(
     '`hds-clipboard` modifier - the modifier must be applied to an element',
     element
   );
 
-  const { text, target, onSuccess, onError } = named;
+  const {
+    text,
+    target,
+    onSuccess,
+    onError,
+  }: {
+    text: string;
+    target: Node;
+    // eslint-disable-next-line @typescript-eslint/no-explicit-any
+    onSuccess: (...args: any[]) => void;
+    // eslint-disable-next-line @typescript-eslint/no-explicit-any
+    onError: (...args: any[]) => void;
+  } = named as {
+    text: string;
+    target: Node;
+    // eslint-disable-next-line @typescript-eslint/no-explicit-any
+    onSuccess: (...args: any[]) => void;
+    // eslint-disable-next-line @typescript-eslint/no-explicit-any
+    onError: (...args: any[]) => void;
+  };
 
-  const onClick = async (event) => {
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  const onClick = async (event: { currentTarget: any }) => {
```

Comment:
The reason why you see an error is because the `element` in the modifier (line 139) is not typed to be an `HTMLElement`
```suggestion
  const onClick = async (event: MouseEvent) => {
```